server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingAction.java

```diff
@@ -73,14 +75,15 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         final TimeValue timeout = request.paramAsTime("master_timeout", getMappingsRequest.masterNodeTimeout());
         getMappingsRequest.masterNodeTimeout(timeout);
         getMappingsRequest.local(request.paramAsBoolean("local", getMappingsRequest.local()));
-        return channel -> client.admin().indices().getMappings(getMappingsRequest, new RestActionListener<>(channel) {
-
-            @Override
-            protected void processResponse(GetMappingsResponse getMappingsResponse) {
-                final long startTimeMs = threadPool.relativeTimeInMillis();
-                // Process serialization on MANAGEMENT pool since the serialization of the raw mappings to XContent can be too slow to
-                // execute on an IO thread
-                threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(
+        final HttpChannel httpChannel = request.getHttpChannel();
+        return channel -> new RestCancellableNodeClient(client, httpChannel).admin().indices().getMappings(getMappingsRequest,
+            new RestActionListener<>(channel) {
```

Comment:
This is quite similar to `DispatchingRestToXContentListener`, with deficiencies on both sides:

- here we have a timeout but `DispatchingRestToXContentListener` doesn't
- `DispatchingRestToXContentListener` checks for cancellation after dispatch but here we don't
- Here we don't have a proper `ToXContentObject`, we call `startObject()` and `endObject()` ourselves, but `DispatchingRestToXContentListener` requires a `ToXContentObject`.

What do you think about moving to `DispatchingRestToXContentListener` here? I think it'd just need a little adapter object to implement the timeout and the outer `XContent` object wrapper, see e.g. `RestClusterStateResponse`.

Reply:
Addressed in https://github.com/elastic/elasticsearch/pull/72234/commits/e2d96baf2c381300d66cbc415623f23dc93f6bc1